### PR TITLE
Support django 1.11.

### PR DESCRIPTION
**Pull request by cahrens**
Updates edx-organizations to use tox (parallelizing testing and quality checks), and then tests with Django 1.11.

@edx/educator-dahlia Can someone review this?

[EDUCATOR-771](https://openedx.atlassian.net/browse/EDUCATOR-771)

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -23,11 +23,11 @@
     ],
     packages=find_packages(exclude=['tests']),
     install_requires=[
-        'django>=1.8,<1.9',
+        'django>=1.8,<2.0',
         'django-model-utils>=1.4.0',
         'djangorestframework>=3.2.0,<3.4.0',
         'djangorestframework-oauth>=1.1.0,<2.0.0',
-        'edx-django-oauth2-provider>=0.5.0,<1.0.0',
+        'edx-django-oauth2-provider>=1.2.0',
```
Comment (cahrens) [inline setup.py]: Without this change, version 0.5.0 was being installed in Travis and would error. Now it is installing 1.2.0 in Travis in passing.

Unfortunately locally it is still installing 0.5.0 and failing, and I haven't been able to figure out why.

Comment (efischer19) [inline setup.py]: If you're having issues locally with tox, try `rm -rf .tox/` from your CWD to reset, I've run into odd caching behavior there before.

Comment (cahrens) [inline setup.py]: Thanks @efischer19, that solved the problem!

image

File: Makefile [hunk 2]
```diff
@@ -1,17 +1,15 @@
...
 requirements:
 	pip install -e .    # Install this package and its dependencies
 	pip install -r test-requirements.txt
 
 test:
-	django-admin.py test organizations --settings=test_settings --with-coverage --cover-package=organizations
-	coverage report
-
-.PHONY: clean, quality, requirements, test
+	tox
```
Comment (cahrens) [inline Makefile]: That's an interesting idea, but I think I will hold off on it. We haven't been doing this in general on the upgrades, and these tests in particular are very fast to run (furthermore, it's an area of code we rarely touch). Thank you though for the suggestion, and the review!

Comment (efischer19) [inline Makefile]: Relevant fact: you can run `tox -e <env>` to run tests in a single environment, such as `tox -e django18` or `tox -e quality`.

Comment (kdmccormick) [inline Makefile]: @cahrens Sounds good, and you're welcome!

**Pull request merged**